src/charm.py

```diff
@@ -687,8 +686,8 @@ def _set_user_created(self, user: MongoDBUser) -> None:
     def _get_mongodb_config_for_user(
         self, user: MongoDBUser, hosts: List[str]
     ) -> MongoDBConfiguration:
-        external_ca, _ = self.tls.get_tls_files(UNIT_SCOPE)
-        internal_ca, _ = self.tls.get_tls_files(APP_SCOPE)
+        external_ca, _ = self.tls.get_tls_files(APP_SCOPE)
+        internal_ca, _ = self.tls.get_tls_files(UNIT_SCOPE)
```

Comment:
I am pretty sure this change should be reverted
```suggestion
        external_ca, _ = self.tls.get_tls_files(UNIT_SCOPE)
        internal_ca, _ = self.tls.get_tls_files(APP_SCOPE)
```

Context: 
- Internal secrets for TLS are used for the MongoDB replicas to communicate with each other with an encrpyted connection, for this, we need all units to have the SAME internal TLS secrets. Since all units should have the **the same** internal secret, it is crucial that this secret is kept consistent across all units. To ensure this, only allow one unit to change this data, so we use "lock" for changing this field, this "lock" is granted to the leader unit. This prevents multiple units from changing the internal secret. APP_DATA works perfect for this because APP_DATA only allowed to be changed by the leader.

- external secrets are used for each unit to communicate outside of the cluster and do not need to be the same for each unit, hence writing to unit data is best here

Reply:
GRRRRRR indeed, VERY GOOD POINT, THANK YOU!!!!!!!!!!!!!!!!!!
I'm also wondering that some tests should have shown an error on this :-( :-( :-( 
(I did this change in desperation when trying to resolve a horrible issue coming from a different source. I can't believe I left it in the code THANK U SO MUCH 4 NOTICING!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!)